composer/datasets/cifar.py

```diff
@@ -24,19 +28,34 @@
     "CIFAR100WebDatasetHparams"
 ]
 
+log = logging.getLogger(__name__)
+
 
 @dataclass
 class CIFAR10DatasetHparams(DatasetHparams, SyntheticHparamsMixin):
     """Defines an instance of the CIFAR-10 dataset for image classification from a local disk.
 
     Args:
         download (bool): Whether to download the dataset, if needed. Default: ``True``.
+        use_ffcv (bool): Whether to use FFCV dataloaders. Default: ``False``.
+        ffcv_dir (str): A directory containing train/val <file>.beton files. If these files don't exist and
+            ``ffcv_write_dataset`` is ``True``, train/val <file>.beton files will be created in this dir. Default: ``"/tmp"``.
+        ffcv_dest_train (str): <file>.beton file that has training samples. Default: ``"cifar_train.beton"``.
+        ffcv_dest_val (str): <file>.beton file that has validation samples. Default: ``"cifar_val.beton"``.
+        ffcv_write_dataset (std): Whether to create dataset in FFCV format (<file>.beton) if it doesn't exist. Default:
+        ``False``.
     """
     download: bool = hp.optional("whether to download the dataset, if needed", default=True)
+    use_ffcv: bool = hp.optional("whether to use ffcv for faster dataloading", default=False)
+    ffcv_dir: str = hp.optional(
+        "A directory containing train/val <file>.beton files. If these files don't exist and ffcv_write_dataset is true, train/val <file>.beton files will be created in this dir.",
```

Comment:
Can we separate out the FFCV dataset writing into a seperate helper function. I'm thinking this function would have a signature like:

```python
def create_ffcv_dataset(dataset: Iterable[Sample], desetination_filepath: str):
    """Iterates through ``dataset`` and converts it into FFCV beton format at filepath ``destination_filepath``."""
```

Then, the user can use a seperate CLI tool to create the dataset, and pass in the ffcv path here.

The reasoning behind separating this out is so 1) ffcv users are not bound to our hparams framework, 2) the dataset hparams is used for loading datasets, and 3) I would like this functionality to be re-usable as we add other datasets in.

Reply:
Added a new create_ffcv_dataset function in composer/datasets/utils.py. The same can be used for imagenet dataset as well.